### PR TITLE
x86: Implement vabs for f/d

### PR DESCRIPTION
**Pull request by BradleyWood**

### CODE REVIEW

**Comment by BradleyWood** [comment]
Jenkins build all

**Comment by BradleyWood** [comment]
@dsouzai Would you mind reviewing when you get a chance?

**Review by dsouzai — COMMENTED** [review]
Code looks ok to me. However, since I'm not a codegen dev, @hzongaro would you mind reviewing the code semantics?

**Comment by BradleyWood** [comment]
Fixed merge conflicts

File: compiler/x/codegen/UnaryEvaluator.cpp [hunk 1]
```diff
@@ -36,15 +36,74 @@
...
+
+   TR::InstOpCode andOpcode = TR::InstOpCode::PANDRegReg;
+   TR::InstOpCode shrOpcode = et == TR::Double ? TR::InstOpCode::PSRLQRegImm1 : TR::InstOpCode::PSRLDRegImm1;
+   OMR::X86::Encoding shrEncoding = shrOpcode.getSIMDEncoding(&cg->comp()->target().cpu, vl);
+   OMR::X86::Encoding andEncoding = andOpcode.getSIMDEncoding(&cg->comp()->target().cpu, vl);
+
+   TR_ASSERT_FATAL(shrEncoding != OMR::X86::Bad, "vabs: No encoding method for shift opcode");
+   TR_ASSERT_FATAL(andEncoding != OMR::X86::Bad, "vabs: No encoding method for and opcode");
+
+   if (vl == TR::VectorLength512)
```
Comment (hzongaro) [inline compiler/x/codegen/UnaryEvaluator.cpp:55]: Being a newbie to the SIMD instructions, it took me a fair bit of time to figure out how the instruction sequences that are being generated implement VABS.  I think a short, high-level description of how this works might help the next person who passes through here.

**Review by hzongaro — COMMENTED** [review]
I think the changes look good.  Just one suggestion regarding comments.

**Comment by BradleyWood** [comment]
I think if resultReg happens to have NaN in it prior to the comparision, the mask may be incorrectly generated. Just to be cautious, I have xor'd it out.

**Comment by BradleyWood** [comment]
Jenkins build all

**Comment by dsouzai** [comment]
https://dev.azure.com/eclipse-omr/omr/_build/results?buildId=6312&view=logs&j=aa2fd561-a338-5537-45ee-49ae22aea958 is likely the same known failure as the jenkins osx build (https://github.com/eclipse/omr/issues/6556).

@hzongaro if everything looks good to you I can merge.

**Review by hzongaro — APPROVED** [review]
Looks good.  Thanks!

**Review by dsouzai — APPROVED** [review]

**Pull request merged**